### PR TITLE
Attendance, Absence and Avg Check-in-time Reports

### LINKED ISSUE #13: Reports View
# REQUIREMENTS
Create view and controller for reports view

# DESIGN
< this view should look like this image >

### PR DESCRIPTION
**Pull request by corps01**
# Description
WHAT:
-Implemented reports controller to generate reports based on Employee, Attendances and Absences models
-Created views for each report alongside their SASS stylesheet.
-Created partial sidebar to access reports

### CODE REVIEW

File: checkClock/app/assets/stylesheets/reports.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        a {
+          color: inherit;
+          cursor:pointer;
+          display: block;
+          height: 100%;
+          text-decoration: none; 
+          width: 100%;
+        }
+    }
+
```
Comment (JuanequeX) [inline checkClock/app/assets/stylesheets/reports.scss]: Change this identation at 2 spaces, please

File: checkClock/app/assets/stylesheets/reports.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
```
Comment (JuanequeX) [inline checkClock/app/assets/stylesheets/reports.scss]: Add a blank line here

File: checkClock/app/controllers/reports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+            if totalEmployeeAbsences == 0
+                next
+            else
+                @EmployeeAbsences.push({ :employeeName => employee.name, :employee_id => employee.id, :listAbsences => listEmployeeAbsences, :totalAbsences => totalEmployeeAbsences }) 
+            end
+
+        end
+    end
+
```
Comment (JuanequeX) [inline checkClock/app/controllers/reports_controller.rb]: possibly in your VS code, you have the system to identation configurate it at 4 spaces, please configure it to 2 spaces to solve this identations

File: checkClock/app/controllers/reports_controller.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+end
```
Comment (JuanequeX) [inline checkClock/app/controllers/reports_controller.rb]: Add a blank line here

File: checkClock/app/views/application/_sidebar.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+          <li class=<%= @current_uri == "/dashboard" ? "active-option" : nil %>><a href="#"><i class="fas fa-tachometer-alt"></i> Dashboard</a></li>
+          <li class=<%= @current_uri == "/employees" ? "active-option" : nil %>><a href="#"><i class="fas fa-users"></i> Employees</a></li>
+          <li class=<%= @current_uri == "/branches" ? "active-option" : nil %>><a href="#"><i class="fas fa-hotel"></i> Branch Offices</a></li>
+        </ul>
+      </div>
+    </div>
+    <div class="logout-container">
+      <%= link_to "Sign out", destroy_admin_session_path, :method => :delete %>
+    </div>
+  </nav>
```
Comment (JuanequeX) [inline checkClock/app/views/application/_sidebar.erb]: Add a blank line here

File: checkClock/app/views/application/_sidebar.erb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        <ul>
+          <li class=<%= @current_uri == "/dashboard" ? "active-option" : nil %>><a href="#"><i class="fas fa-tachometer-alt"></i> Dashboard</a></li>
+          <li class=<%= @current_uri == "/employees" ? "active-option" : nil %>><a href="#"><i class="fas fa-users"></i> Employees</a></li>
+          <li class=<%= @current_uri == "/branches" ? "active-option" : nil %>><a href="#"><i class="fas fa-hotel"></i> Branch Offices</a></li>
+        </ul>
+      </div>
+    </div>
+    <div class="logout-container">
+      <%= link_to "Sign out", destroy_admin_session_path, :method => :delete %>
+    </div>
```
Comment (JuanequeX) [inline checkClock/app/views/application/_sidebar.erb]: Same here, check your identation

File: checkClock/app/views/reports/absences.erb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+  <tbody>
+       <% employee[:listAbsences].each do |absence| %>
+      <tr>
+      <td><%= absence.date_absence.strftime("%d/%m/%Y") %></td> 
+      </tr>
+      <% end %>
+  </tbody>
+</table>
+<% end %>
```
Comment (JuanequeX) [inline checkClock/app/views/reports/absences.erb]: Add a blank line hera and review this identation

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by dany-herz42 — APPROVED** [review]

**Pull request merged**